packages/uicore/src/components/DateInput/DateInput.tsx

```diff
@@ -206,9 +206,9 @@ export const DateInput: React.FC<DateInputProps> = props => {
   // On Change with CAlender
   const onChangeDate = (selectedDate: Date) => {
     if (onChange) {
-      onChange(selectedDate.getTime().toString())
+      onChange(selectedDate?.getTime().toString())
```

Comment:
If type is Date then should not need optional check. This might be giving issues when selectedDate is undefined / null. Don't we need to have some fallback mechanism to handle undefined / null.